### PR TITLE
[Agent] Parsing the statement id of the mysql protocol #3590

### PR DESCRIPTION
**Pull request by yuanchaoa**
### This PR is for:

- Agent

#### Checklist
- [ ] Added unit test.
#### Backport to branches
- main

### CODE REVIEW

File: agent/resources/test/flow_generator/mysql/mysql-database.result [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-MysqlInfo { msg_type: Request, is_tls: false, protocol_version: 10, server_version: "", server_thread_id: 0, command: 1, context: "", response_code: 0, error_code: None, affected_rows: 0, error_message: "", status: Ok, rrt: 0, trace_id: None } is_mysql: false
+MysqlInfo { msg_type: Other, is_tls: false, protocol_version: 10, server_version: "", server_thread_id: 0, command: 0, context: "", response_code: 0, error_code: None, affected_rows: 0, error_message: "", status: Ok, rrt: 0, statement_id: 0, trace_id: None } is_mysql: false
+MysqlInfo { msg_type: Other, is_tls: false, protocol_version: 10, server_version: "", server_thread_id: 0, command: 0, context: "", response_code: 0, error_code: None, affected_rows: 0, error_message: "", status: Ok, rrt: 0, statement_id: 0, trace_id: None } is_mysql: false
+MysqlInfo { msg_type: Response, is_tls: false, protocol_version: 10, server_version: "", server_thread_id: 0, command: 0, context: "", response_code: 0, error_code: None, affected_rows: 0, error_message: "", status: Ok, rrt: 0, statement_id: 131072, trace_id: None } is_mysql: false
```
Comment (yuanchaoa) [inline agent/resources/test/flow_generator/mysql/mysql-database.result:3]: statement_id 解析有的时候会误报

Comment (yuanchaoa) [reply agent/resources/test/flow_generator/mysql/mysql-database.result:3]: @sharang

File: message/flow_log.proto [hunk 2]
```diff
@@ -213,6 +213,7 @@ message AppProtoLogsData {
 
     uint32 row_effect = 16;
     uint32 direction_score = 17;
+    uint32 statement_id = 18;
```
Comment (yuanchaoa) [inline message/flow_log.proto:216]: @lzf575  注意这里新添加了一个字段

File: agent/src/flow_generator/protocol_logs/sql/mysql.rs [hunk 3]
```diff
@@ -171,6 +181,12 @@ impl MysqlInfo {
             self.trace_id = extra_sql_trace_id(self.context.as_str(), t);
         }
     }
+
+    fn statement_id(&mut self, payload: &[u8]) {
+        if payload.len() >= STATEMENT_ID_LEN {
+            self.statement_id = u32::from_le_bytes(payload[..STATEMENT_ID_LEN].try_into().unwrap());
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/sql/mysql.rs:187]: 有个`public::bytes::read_u32_le`

**Review by sharang — APPROVED** [review]

**Pull request merged**